### DISCUSSION TITLE
PostgreSQL SSL connection

### DISCUSSION THREAD

**Discussion by legaljustus** [Budibase/budibase#5477] [Help]
Hi! 

I'm trying to connect my budibase to my PostgreSQL database with SSL. However when i copy the text of my .crt file into the Ca field i get the following error "connection requires a valid client certificate". I'm certainly no TLS / SSL expert so it might just be that i'm putting in the wrong stuff. But i have seen this kind of thing work on other platforms with just a file upload of your cert, might be easier to use right?

Below you can see how i put it in (i took out the real cert)

FYI the devops guy im doing this project with uses self signed certificates 

Thanks a lot, in advance, for any help!

![image](https://user-images.githubusercontent.com/83215912/163967172-8fafadf6-7c57-4a03-b90c-fe3157082eaf.png)

**Comment by melohagan** [comment]
Hi @legaljustus 

That error message is being relayed from PostgreSQL, so I would double check that the certificate is correct. 

This may help: https://serverfault.com/questions/663682/how-to-specify-a-client-certificate-to-psql

**Reply by legaljustus** [reply]
Thanks for the help @melohagan,

In the link you provided they are talking about specifying a file path. I currently don't have the ssl cert on my budibase instance, I copied it and just put it in the field as a string. Should i host the .cert file on my server and then put in a file path in this field?

Thanks so much for the help!

**Reply by melohagan** [reply]
Hey @legaljustus No, I think pasting the contents of the cert into that field is correct. 

What I was suggesting is that you use the command from that post, outside of budibase, to verify that the certificate is definitely correct.

**Reply by legaljustus** [reply]
Ah okay thanks for the clarification! I'll try that asap!

**Comment by muscovitebob** [comment]
Is private key auth actually supported? I get this message if I put the private key into the CA field, however the CA cert is a separate item. Should there not be a separate field for the private key?
